lib/querystring.js

```diff
@@ -5,6 +5,18 @@
 const QueryString = exports;
 const Buffer = require('buffer').Buffer;
 
+// This constructor is used to store parsed query string values. Instantiating
+// this is faster than explicitly calling `Object.create(null)` to get a
+// "clean" empty object
+function ParsedQueryString() {}
+ParsedQueryString.prototype = Object.create(null, {
+  constructor: {
```

Comment:
what's with the `constructor` if I might ask? why bothering?

Also, I think you still need to return an `instanceof Object` to avoid breaking code using this module.